src/account/update.rs

```diff
@@ -40,7 +40,7 @@ impl AccountHandle {
         &self,
         addresses_with_unspent_outputs: Vec<AddressWithUnspentOutputs>,
         unspent_outputs: Vec<OutputData>,
-        spent_outputs: Vec<OutputId>,
+        spent_or_not_synced_output_ids: Vec<OutputId>,
         spent_or_not_synced_output_responses: Vec<OutputResponse>,
```

Comment:
Do we even need the ids at all ? Seems to me we could achieve the same without it at all

Reply:
We need the ids because if the output is pruned and the node can't return the metadata, then we can't set it as spent if we just work with the metadata we get

Reply:
Can't we imagine anything like a HashMap of Option OutputResponse or whatever? Fine if we can't, but that double loop is a bit annoying

Reply:
Yes HashMap with Option should work